src/ast/nodes/IfStatement.ts

```diff
@@ -45,33 +50,41 @@ export default class IfStatement extends StatementBase {
 	}
 
 	render(code: MagicString, options: RenderOptions) {
-		const testValue = this.test.getValue();
+		const testValue = this.hasUnknownTestValue ? UNKNOWN_VALUE : this.test.getValue();
 		if (
 			!this.context.treeshake ||
```

Comment:
I'm not quite sure I follow why this check is being done since it seems like we are performance treeshaking in this block?

Reply:
You are absolutely right, this check is totally unnecessary as in the no-treeshaking-case, all nodes are included a-priori anyway. I have also adjusted the few other places where such a check was made so that the no-treeshaking logic is now very localized indeed.